atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -2428,6 +2429,71 @@ public void setsRequestedCommitLocksCountCorrectly_serializableLockLevelMigratio
         assertThat(commitLockInfo.rowCommitLocksRequested()).isEqualTo(1 + 1);
     }
 
+    @Test
+    public void emitExhaustiveMetricWhenReadingExistingValue() {
+        putCellsInTable(List.of(TEST_CELL), TABLE_SWEPT_THOROUGH);
+
+        long transactionTs = timelockService.getFreshTimestamp();
+        LockImmutableTimestampResponse res = timelockService.lockImmutableTimestamp();
+        Transaction transaction =
+                getSnapshotTransactionWith(timelockService, () -> transactionTs, res, PreCommitConditions.NO_OP, false);
+
+        long exhaustiveReadsBeforeGet = getReadCounter(true).getCount();
+        long nonExhaustiveReadsBeforeGet = getReadCounter(false).getCount();
```

Comment:
ugh, I suspect this is because the test shares some state? and/or there's some reason we can't "just" assert 0 before, and then exhaustive=1 and non-exhaustive=0 after that!